galaxy/tools/deps/mulled/get_tests.py

```diff
@@ -0,0 +1,263 @@
+    else:
+        for requirement in requirements:
+            if requirement.split()[0] == 'perl':
+                package_tests['import_lang'] = 'perl -e'
+                break
+            # elif ... :
+                # other languages if necessary ... hopefully python and perl should suffice though
+        else: # python by default
+            package_tests['import_lang'] = 'python -c'
+    return package_tests
+    
+
+def get_runtest(file):
+    """
+    Gets tests from a run_test.sh file
+
+    """
+    package_tests = {}
+    package_tests['commands'] = [file.read().replace('\n', ' && ')]
+    return package_tests
+
+
+def get_anaconda_url(container):
+    """
+    Downloading tarball from anaconda for test
+    """
+    name = container.replace('--', ':').split(':') # list consisting of [name, version, (build, if present)]
+    return "https://anaconda.org/bioconda/%s/%s/download/linux-64/%s.tar.bz2" % (name[0], name[1], '-'.join(name))
+
+
+def get_test_from_anaconda(url):
+    """
+    Given the URL of an anaconda tarball, returns tests
+    >>> get_test_from_anaconda('https://anaconda.org/bioconda/samtools/1.3.1/download/linux-64/samtools-1.3.1-5.tar.bz2')
+    {'commands': ['samtools --help'], 'import_lang': 'python -c'}
+    """
+    r = requests.get(url)
+    
+    try:
+        tarball = tarfile.open(mode="r:bz2", fileobj=BytesIO(r.content))
+    except tarfile.ReadError:
+        return None
+
+    try:
+        metafile = tarball.extractfile('info/recipe/meta.yaml')
+    except (tarfile.ReadError, KeyError, TypeError):
+        pass
+    else:
+        package_tests = get_commands_from_yaml(metafile)
+        if package_tests:
+            return package_tests
+
+    # this part is probably unnecessary, but some of the older tarballs have a testfile with .yaml.template ext
+    # try:
+    #     metafile = tarball.extractfile('info/recipe/meta.yaml.template')
+    # except (tarfile.ReadError, KeyError, TypeError):
+    #     pass
+    # else:
+    #     package_tests = get_commands_from_yaml(metafile)
+    #     if package_tests: 
+    #         return package_tests
+
+    # if meta.yaml was not present or there were no tests in it, try and get run_test.sh instead
+    try: 
+        run_test = tarball.extractfile('info/recipe/run_test.sh')
+        return get_runtest(run_test)
+    except KeyError:
+        return None
+        logging.info("run_test.sh file not present.")
+
+
+def find_anaconda_versions(name):
+    r = requests.get("https://anaconda.org/bioconda/%s/files" % name)
+    urls = []
+    for line in r.text.split('\n'):
+        if 'download/linux' in line:
+            urls.append(line.split('"')[1])
+    return urls
+
+def open_recipe_file(file):
+	if RECIPES_REPO_PATH:
+		return open('%s/%s' % (RECIPES_REPO_PATH, file))
+	else: # if no clone of the repo is available locally, download from GitHub
+		r = requests.get('https://raw.githubusercontent.com/bioconda/bioconda-recipes/master/%s' % recipes)
```

Comment:
bioconda as parameter.
I would fix at first the tests and include more and later if this all works make bioconda a variable - this will hopefully catch many errors.